### PR TITLE
better tests for turf-point-on-line

### PR DESCRIPTION
**Pull request by morganherlocker**
This PR updates test for turf-point-on-line to verify several bugs have been handled around [line snapping edge cases](https://github.com/Turfjs/turf/issues/388).

### CODE REVIEW

File: packages/turf-point-on-line/test/test.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+var lineDistance = require('turf-line-distance');
+var along = require('turf-along');
+
+test('turf-point-on-line - first point', function (t) {
+    var line = linestring([[-122.45717525482178,37.72003306385638],[-122.45717525482178,37.718242366859215]]);
+    var pt = point([-122.45717525482178,37.72003306385638]);
+
+    var snapped = pointOnLine(line, pt);
+
+    t.equal(pt.geometry.coordinates.join(','), snapped.geometry.coordinates.join(','), 'pt on start does not move');
```
Comment (tmcw) [inline packages/turf-point-on-line/test/test.js]: why does this join the points instead of using t.deepEqual?
